chia/wallet/wallet_nft_store.py

```diff
@@ -87,22 +124,29 @@ async def save_nft(self, wallet_id: uint32, did_id: Optional[bytes32], nft_coin_
                     int(nft_coin_info.latest_height),
                 ),
             )
-            await cursor.close()
             # Rotate the old removed NFTs, they are not possible to be reorged
-            await (
-                await conn.execute(
-                    "DELETE FROM users_nfts WHERE removed_height is not NULL and removed_height<?",
-                    (int(nft_coin_info.latest_height) - REMOVE_BUFF_BLOCKS,),
-                )
-            ).close()
+            await conn.execute(
+                "DELETE FROM users_nfts WHERE removed_height is not NULL and removed_height<?",
+                (int(nft_coin_info.latest_height) - REMOVE_BUFF_BLOCKS,),
+            )
+
+    async def count(self, wallet_id: Optional[uint32] = None, did_id: Optional[bytes32] = None) -> int:
+        sql = "SELECT COUNT(nft_id) FROM users_nfts WHERE removed_height is NULL"
+        if wallet_id is not None:
+            sql += f" AND wallet_id={wallet_id}"
+        if did_id:
+            sql += f" AND did_id='{did_id}'"
+
+        async with self.db_wrapper.reader_no_transaction() as conn:
+            rows = list(await conn.execute_fetchall(sql))
```

Comment:
there is free function called `execute_fetchone()` which does this in one line, and returns an `Optional[Row]`.

```
from chia.util.db_wrapper import execute_fetchone
```